wandb/sdk/artifacts/artifact.py

```diff
@@ -1686,6 +1686,78 @@ def download(
         root = root or self._default_root()
```

Comment:
Nit: then `root` should not be nullable in `_run_artifact_download`, `communicate_download_artifact`, `_download`.